### PR TITLE
Flutter for React Native Developers

### PR DESCRIPTION
**Pull request by sanketsahusoft**
A comprehensive document for React Native developers to learn Flutter.

### CODE REVIEW

**Comment by mjohnsullivan** [comment]
Please sign the cla

**Comment by sanketsahusoft** [comment]
I signed it.

File: flutter-for-react-native.md [hunk 1]
```diff
@@ -0,0 +1,2007 @@
...
+
+This document is for React Native developers looking to apply their existing React Native knowledge to build mobile apps with Flutter. If you understand the fundamentals of the React Native framework, then you can use this document as a jump start to your Flutter development. 
+
+
+* TOC Placeholder
+{:toc}
+
+## A Brief Introduction to Dart for JavaScript Developers
+
+While React Native uses Javascript, Flutter uses a language called Dart. [Dart](https://www.dartlang.org/) is an open-source, scalable programming language, for building web, server, and mobile apps. It is an object-oriented, single inheritance language that uses a C-style syntax that is AOT-compiled into native and also transcompiles optionally into JavaScript. It supports interfaces, abstract classes and strong mode.
```
Comment (Hixie) [inline flutter-for-react-native.md]: "and strong mode" seems a bit weird. Maybe "and types"?

File: flutter-for-react-native.md [hunk 2]
```diff
@@ -0,0 +1,2007 @@
...
+print('Hello World');
+{% endprettify %}
+
+Try it out in [DartPad](https://dartpad.dartlang.org/cf9e652f77636224d3e37d96dcf238e5).
+
+
+### Variables
+#### Creating and Assigning Variables
+
+While JavaScript variables cannot be typed, Dart variables are optionally typed but it is a good practice to use typed variables. In [Dart 2](https://www.dartlang.org/dart-2), types are mandatory. With both static and runtime type checks, it has a sound type system. This type system enables better tooling, as well as earlier feedback when you write code.
```
Comment (Hixie) [inline flutter-for-react-native.md]: Types are not mandatory in Dart 2.

File: flutter-for-react-native.md [hunk 3]
```diff
@@ -0,0 +1,2007 @@
...
+Consider this example in Javascript that makes use of the Generator function:
+
+<!-- skip -->
+{% prettify javascript %}
+// JavaScript
+var sum=0;
+function* sumStream () {
+  var n = 1;
+  while (true) {
+    yield sum=sum+n++;
```
Comment (Hixie) [inline flutter-for-react-native.md]: this is more equivalent to the sync*/Iterable pattern than the async*/Stream pattern.

I would probably not mention streams, to be honest. You don't use them much in Flutter apps at first.

File: flutter-for-react-native.md [hunk 4]
```diff
@@ -0,0 +1,2007 @@
...
+                'Hello World',
+              ),
+            ),
+          ],
+        ),
+      )
+    );
+  }
+}
+{% endprettify %}
```
Comment (Hixie) [inline flutter-for-react-native.md]: these two codes snippets aren't equivalent at all. We can dramatically simplify the Flutter version to make it more equivalent to the RN version.

File: flutter-for-react-native.md [hunk 5]
```diff
@@ -0,0 +1,2007 @@
...
+  style={{
+    flex: 1,
+    flexDirection: "column",
+    justifyContent: “space-between”,
+    alignItems: “center”
+  }}
+>
+{% endprettify %}
+
+In Flutter, layout is determined by a combination of the type of Widget that you choose and its layout and style properties. Contray to how it is done in React Native, where most of the layout can be done with the props that are passed to a specific component, in Flutter most of the layout is done by the use of widgets that are specifically designed to provide layout to the underlying widget.
```
Comment (Hixie) [inline flutter-for-react-native.md]: Contray->Contrary

File: flutter-for-react-native.md (hunk 1 above)
Comment (wmleler) [reply flutter-for-react-native.md]: Or "and strong types".

Also, spell it "JavaScript" consistently. And remove the comma at the end of "scalable programming language,"

File: flutter-for-react-native.md [hunk 6]
```diff
@@ -0,0 +1,2007 @@
...
+{% endprettify %}
+
+The below code snippet shows how the state implements the build function and also how `setState` function is used to rebuild (recall the build function of the stateful `_MyStatefulWidgetState` widget).
+
+<!-- skip -->
+{% prettify dart %}
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+  bool showtext=true;
+  bool toggleState=true;
+  Timer t2;
```
Comment (Hixie) [inline flutter-for-react-native.md]: this timer isn't used

**Comment by sanketsahusoft** [comment]
I have made the changes and created a new PR [here](https://github.com/flutter/website/pull/801) due to complications in the CLA for this PR. 
We can discuss the rest there.

**Pull request closed without merging**